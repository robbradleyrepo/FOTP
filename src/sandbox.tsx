/* eslint-disable simple-import-sort/imports */

// Import for side effects (`configureXYZ(...)`)
import "../pages/_app";

import Sandbox from "@sss/apollo/testing";
import * as ecommerce from "@sss/ecommerce/testing";

import PageProviders from "./ui/providers";
import introspectionResultData from "../generated/config/fragment-types.json";
import { I18N } from "./config";

export const anonymous = new Sandbox({
  Wrapper: PageProviders,
  globalMocks: [
    ecommerce.mockedResponses.checkout.none,
    ecommerce.mockedResponses.collection.core,
  ],
  introspectionResultData,
  locale: I18N.fallbackLocale,
});

export default anonymous;
