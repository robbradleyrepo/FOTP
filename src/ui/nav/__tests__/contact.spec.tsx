import { advanceTo } from "jest-date-mock";
import React from "react";

import { anonymous } from "../../../sandbox";
import Contact from "../contact";

jest.useFakeTimers();

const businessHours = [
  "2021-01-01T09:00:00.000-08:00", // Friday (PST)
  "2021-01-01T16:59:59.999-08:00", // Friday (PST)
  "2021-07-01T09:00:00.000-07:00", // Thursday (PDT)
  "2021-07-01T16:59:59.999-07:00", // Thursday (PDT)
];
const nonBusinessHours = [
  "2021-01-02T09:00:00.000-08:00", // Saturday (PST)
  "2021-01-03T09:00:00.000-08:00", // Sunday (PST)
  "2021-01-01T08:59:59.999-08:00", // Friday (PST)
  "2021-01-01T17:00:00.000-08:00", // Friday (PST)
  "2021-07-01T08:59:59.999-07:00", // Thursday (PDT)
  "2021-07-01T17:00:00.000-07:00", // Thursday (PDT)
];

describe("<Contact />", () => {
  it("should render the customer service phone number during business hours", async () => {
    for (let i = 0; i < businessHours.length; i += 1) {
      advanceTo(new Date(businessHours[i]));

      const { component } = await anonymous.component({
        Component: <Contact />,
      });

      await expect(component.findByRole("link")).resolves.toMatchSnapshot();
    }
  });

  it("should render the customer service email address outside of business hours", async () => {
    for (let i = 0; i < nonBusinessHours.length; i += 1) {
      advanceTo(new Date(nonBusinessHours[i]));

      const { component } = await anonymous.component({
        Component: <Contact />,
      });

      await expect(component.findByRole("link")).resolves.toMatchSnapshot();
    }
  });
});
