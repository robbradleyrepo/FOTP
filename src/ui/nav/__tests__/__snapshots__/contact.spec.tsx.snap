// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render the customer service email address outside of business hours 1`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service email address outside of business hours 2`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service email address outside of business hours 3`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service email address outside of business hours 4`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service email address outside of business hours 5`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service email address outside of business hours 6`] = `
<a
  class="contact___StyledMotionA2-sc-v3d6xv-2 cuZFfi"
  href="mailto:hello@fotp.com"
>
  Questions?
   
  Email us:
  <span
    class="contact___StyledSpan2-sc-v3d6xv-3 izgUJd"
  >
    hello@fotp.com
  </span>
</a>
`;

exports[`<Contact /> should render the customer service phone number during business hours 1`] = `
<a
  class="contact___StyledMotionA-sc-v3d6xv-0 gJPoce"
  href="tel:+13239225737"
>
  Questions?
   
  Call us:
  <span
    class="contact___StyledSpan-sc-v3d6xv-1 cOLMzg"
  >
    323-922-5737
  </span>
</a>
`;

exports[`<Contact /> should render the customer service phone number during business hours 2`] = `
<a
  class="contact___StyledMotionA-sc-v3d6xv-0 gJPoce"
  href="tel:+13239225737"
>
  Questions?
   
  Call us:
  <span
    class="contact___StyledSpan-sc-v3d6xv-1 cOLMzg"
  >
    323-922-5737
  </span>
</a>
`;

exports[`<Contact /> should render the customer service phone number during business hours 3`] = `
<a
  class="contact___StyledMotionA-sc-v3d6xv-0 gJPoce"
  href="tel:+13239225737"
>
  Questions?
   
  Call us:
  <span
    class="contact___StyledSpan-sc-v3d6xv-1 cOLMzg"
  >
    323-922-5737
  </span>
</a>
`;

exports[`<Contact /> should render the customer service phone number during business hours 4`] = `
<a
  class="contact___StyledMotionA-sc-v3d6xv-0 gJPoce"
  href="tel:+13239225737"
>
  Questions?
   
  Call us:
  <span
    class="contact___StyledSpan-sc-v3d6xv-1 cOLMzg"
  >
    323-922-5737
  </span>
</a>
`;
